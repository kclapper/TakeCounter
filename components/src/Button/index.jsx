import React from 'react';
import { useRef, useEffect } from 'react';

import bootstrap from 'bootstrap/dist/js/bootstrap.bundle';

function Button({ onClick, tooltipPlacement, tooltip, className, children}) {

  const element = useRef(null);

  useEffect(() => {
    if (element.current !== null && Boolean(tooltip)) {
      new bootstrap.Tooltip(element.current, {
        delay: {
          show: 1000,
          hide: 200
        }
      });
    }
  }, [tooltip, tooltipPlacement, element]);

  return <button ref={ element }
                 className={ className || 'btn btn-primary m-1' }
                 type='button'
                 onClick={ onClick }
                 data-bs-toggle="tooltip"
                 data-bs-placement={ tooltipPlacement || "top" }
                 data-bs-title={ tooltip }>
           { children }
         </button>
}

export default Button;
