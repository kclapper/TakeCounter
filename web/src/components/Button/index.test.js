import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { expect, it } from '@jest/globals';

import Button from './index';

function matchSnapshot(comp) {
  const tree = renderer.create(comp).toJSON();
  expect(tree).toMatchSnapshot();
}

function setup(comp) {
  return {
    user: userEvent.setup(),
    ...render(comp)
  };
}

it('Renders with no props', () => {
  matchSnapshot(<Button />)
});

it('Renders with text child', () => {
  matchSnapshot(<Button>
                  Click Me!
                </Button>)
});

it('Renders tooltip', async () => {
  const {
    getByText,
    findByText,
  } = setup(<Button tooltip='test'>
              Click Me!
            </Button>);

  fireEvent.mouseOver(getByText('Click Me!'));

  expect(await findByText('test')).toBeVisible();
});
