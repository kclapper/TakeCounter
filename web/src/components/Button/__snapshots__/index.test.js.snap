// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with no props 1`] = `null`;

exports[`Renders with text child 1`] = `null`;
